vm/systemSmartContracts/auction.go

```diff
@@ -635,12 +693,21 @@ func (s *stakingAuctionSC) unStake(args *vmcommon.ContractCallInput) vmcommon.Re
 		return vmcommon.UserError
 	}
 
+	err = s.eei.UseGas(s.gasCost.MetaChainSystemSCsCost.UnStake)
+	if err != nil {
+		return vmcommon.OutOfGas
+	}
+
 	blsKeys, err := getBLSPublicKeys(registrationData, args)
 	if err != nil {
 		return vmcommon.UserError
 	}
 
 	for _, blsKey := range blsKeys {
+		err = s.eei.UseGas(s.gasCost.BaseOperationCost.DataCopyPerByte * uint64(len(blsKey)))
+		if err != nil {
+			return vmcommon.OutOfGas
+		}
 		_, err := s.executeOnStakingSC([]byte("unStake@" + hex.EncodeToString(blsKey) + "@" + hex.EncodeToString(registrationData.RewardAddress)))
 		if err != nil {
 			log.Debug("blsKey unStaking", "error", err)
```

Comment:
Could it happen that the executeOnStakingSC does not unstake and the result of the SC is OK? What should happen next?

Reply:
added change to treat vmoutput as well. Good finding.

Reply:
ok